execution-engine/engine-tests/benches/transfer_bench.rs

```diff
@@ -0,0 +1,147 @@
+use casperlabs_engine_tests::support::test_support::{
+    LmdbWasmTestBuilder, WasmTestResult, DEFAULT_BLOCK_TIME, STANDARD_PAYMENT_CONTRACT,
+};
+use contract_ffi::value::account::PublicKey;
+use contract_ffi::value::U512;
+use engine_core::engine_state::EngineConfig;
+use engine_core::engine_state::MAX_PAYMENT;
+use engine_storage::global_state::lmdb::LmdbGlobalState;
+use std::collections::HashMap;
+use std::fs::{self, DirEntry};
+use std::io;
+use std::path::Path;
+use tempfile::TempDir;
+
+const GENESIS_ADDR: [u8; 32] = [1; 32];
+
+fn engine_with_payments() -> EngineConfig {
+    EngineConfig::new().set_use_payment_code(true)
+}
+
+fn bootstrap(accounts: &[PublicKey]) -> (WasmTestResult<LmdbGlobalState>, TempDir) {
+    println!("Creating {} accounts...", accounts.len());
+    let accounts_bytes: Vec<Vec<u8>> = accounts
+        .iter()
+        .map(|public_key| public_key.value().to_vec())
+        .collect();
+    let amount = U512::from(1);
+
+    let data_dir = TempDir::new().expect("should create temp dir");
+    let result = LmdbWasmTestBuilder::new_with_config(&data_dir.path(), engine_with_payments())
+        .run_genesis(GENESIS_ADDR, HashMap::new())
+        .exec_with_args(
+            GENESIS_ADDR,
+            STANDARD_PAYMENT_CONTRACT,
+            (U512::from(MAX_PAYMENT),),
+            "create_accounts.wasm",
+            (accounts_bytes, amount), //args
+            DEFAULT_BLOCK_TIME,       // blocktime
+            [1; 32],                  // deploy_hash
+        )
+        .expect_success()
+        .commit()
+        .finish();
+
+    // println!("Bootstrap finished in {}ms", now.elapsed().as_millis());
+    (result, data_dir)
+}
+
+fn visit_dirs(dir: &Path, cb: &dyn Fn(&DirEntry)) -> io::Result<()> {
```

Comment:
```suggestion
fn visit_dirs(dir: &Path, cb: &impl Fn(&DirEntry)) -> io::Result<()> {
```

...and I _belieeeve_ then we are not using dynamic dispatch...